x/concentrated-liquidity/pool.go

```diff
@@ -8,17 +8,21 @@ import (
 
 	"github.com/osmosis-labs/osmosis/v12/osmomath"
 	cltypes "github.com/osmosis-labs/osmosis/v12/x/concentrated-liquidity/types"
-	"github.com/osmosis-labs/osmosis/v12/x/gamm/types"
+	gammtypes "github.com/osmosis-labs/osmosis/v12/x/gamm/types"
 )
 
-func (k Keeper) CreateNewConcentratedLiquidityPool(ctx sdk.Context, poolId uint64, denom0, denom1 string, currSqrtPrice sdk.Dec, currTick sdk.Int) (Pool, error) {
-	poolAddr := types.NewPoolAddress(poolId)
+// TODO: spec, tests, implementation
+func (k Keeper) InitializePool(ctx sdk.Context, pool gammtypes.PoolI, creatorAddress sdk.AccAddress) error {
+	return nil
+}
```

Comment:
let's remove this for now?

Reply:
This must be defined because the interface for the concentrated-liquidity module field in swap router keeper requires it